app/view/sdl/ResetTimeoutPopUp.js

```diff
@@ -0,0 +1,392 @@
+     * on the popUp
+     */
+    resetTimeOutLabel: function () {
+        let message = '';
+        for (let key in this.resetTimeoutRPCs) {
+            message = `${message} ${this.resetTimeoutRPCs[key].method} : ${this.resetTimeoutRPCs[key].timeoutSeconds}\n`
+        }
+        this.set('timeoutString', message);
+    },
+
+    /*
+     * resetTimeout function. sends to SDL OnResetTimeout
+     */
+    resetTimeout: function () {
+        if (this.resetPeriod > 1000) {
+            this.set('resetPeriod', this.defaultTimeout);
+            SDL.ControlButtons.set('resetPeriodInput', this.defaultTimeout * 1000);
+            document.getElementById('resetPeriodInput').value = this.defaultTimeout * 1000;
+            return
+        }
+
+        reset = () => {
+            const requestID = Object.keys(this.resetTimeoutRPCs)[0];
+            if(this.resetTimeoutRPCs[requestID].count === false) return;
+            const originalTimeout = this.resetTimeoutRPCs[requestID].originalTimeout;
+            const resetPeriod = originalTimeout > this.resetPeriod ? originalTimeout : this.resetPeriod;
+            this.resetTimeoutRPCs[requestID].timeoutSeconds = resetPeriod;
+            if (this.resetTimeoutRPCs[requestID].resetTimeoutCallback !== undefined) {
+                this.resetTimeoutRPCs[requestID].resetTimeoutCallback(this.resetTimeoutRPCs[requestID].timeoutSeconds * 1000);
+            }
+            this.resetTimeOutLabel();
+            FFW.BasicCommunication.OnResetTimeout(parseInt(requestID), this.resetTimeoutRPCs[requestID].method, resetPeriod * 1000);
+        }
+
+        const length = this.getPRCsLength();
+        if (1 < length) {
+            this.resetMoreThanOneTimeout();
+            return;
+        }
+        reset();
+    },
+
+    resetTimeoutSpecificRpc(rpcName) {
+        for(const [requestID, value] of Object.entries(this.resetTimeoutRPCs)) {
+            if(value.method === rpcName) {
+                const originalTimeout = value.originalTimeout;
+                const resetPeriod = originalTimeout > this.resetPeriod ? originalTimeout : this.resetPeriod;
+                value.timeoutSeconds = resetPeriod;
+                if (value.resetTimeoutCallback !== undefined) {
+                    value.resetTimeoutCallback(value.timeoutSeconds * 1000);
+                }
+                this.resetTimeOutLabel();
+                FFW.BasicCommunication.OnResetTimeout(parseInt(requestID), value.method, value.timeoutSeconds * 1000);
+            }
+        }
+    },
+
+    /*
+     * timerHandler function. deactivate popup after timeout is expired
+     */
+    timerHandler: function () {
+        let length = this.getPRCsLength();
+        if (1 < length) {
+            timeoutExpired = [];
+            const TIME_OUT_EXPIRATION_SECONDS = 0;
+            for (let [key, value] of Object.entries(this.resetTimeoutRPCs)) {
+                if (TIME_OUT_EXPIRATION_SECONDS === value.timeoutSeconds) {
```

Comment:
```suggestion
                if (TIME_OUT_EXPIRATION_SECONDS >= value.timeoutSeconds) {
```

it cannot be assumed that all timeouts will be even seconds.

I am concerned this is major issue with the implementation. How can we make this timer more granular?

If we just make the change I suggest here, and app sends request with timeout = 9001, timeout callback won't be called for 10000ms, meaning there is high chance Core might not receive this response before sending GENERIC_ERROR.

Reply:
Done 89c9273c47dbf5941dd4855ccf01e3809b020348